### DISCUSSION TITLE
Documentation for `query` and `utils`

### DISCUSSION THREAD

**Discussion by jamestrew** [nvim-treesitter/nvim-treesitter#3017] [Q&A] [answered]
Hi folks,
I'm trying to find some documentation for `nvim-treesitter.utils` (not `ts_utils`) and `nvim-treesitter.query` but I haven't had any luck. 

Does any documentation exist on these topics and if it does, where can I find it? Or do I just dive into the code?

Thanks

**Comment by kyazdani42** [comment]
nvim-treesitter.utils contains mainly internal stuff for nvim-treesitter, so there is not documentation for it since it's not a public api.
Feel free to read the code if you wish to use it

**Comment by clason** [comment, marked as answer]
Also, much of it is in the process of being upstreamed (https://github.com/neovim/neovim/pull/18232), which should also improve the documentation.
